src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -30,16 +33,30 @@ internal ChainedHeaderTree CreateChainedHeaderTree()
                 return this.ChainedHeaderTree;
             }
 
-            public ChainedHeader ExtendAChain(int count, ChainedHeader chainedHeader = null)
+            internal Target ChangeDifficulty(ChainedHeader header, int difficultyAdjustmentDivisor)
             {
+                BigInteger newTarget = header.Header.Bits.ToBigInteger();
+                newTarget = newTarget.Divide(BigInteger.ValueOf(difficultyAdjustmentDivisor)); 
+                return new Target(newTarget);
+            }
+
+            public ChainedHeader ExtendAChain(int count, ChainedHeader chainedHeader = null, int difficultyAdjustmentDivisor = 1)
+            {
+                if (difficultyAdjustmentDivisor == 0) throw new ArgumentException("Divisor cannot be 0");
                 ChainedHeader previousHeader = chainedHeader ?? new ChainedHeader(this.Network.GetGenesis().Header, this.Network.GenesisHash, 0);
 
                 for (int i = 0; i < count; i++)
                 {
                     BlockHeader header = this.Network.Consensus.ConsensusFactory.CreateBlockHeader();
                     header.HashPrevBlock = previousHeader.HashBlock;
-                    header.Bits = previousHeader.Header.Bits - 1000; // just increase difficulty.
+                    header.Bits = difficultyAdjustmentDivisor == 1 
+                                        ? previousHeader.Header.Bits 
+                                        : this.ChangeDifficulty(previousHeader, difficultyAdjustmentDivisor);
+                    header.Nonce = (uint)new Random().Next(1, 100000);
```

Comment:
Once in awhile the test will fail just by chance of generating the same nonce. It would be better to keep a global variable did you increment with interlocked

Reply:
better to keep a global variable `did you increment with interlocked`. Can you elaborate, I am not understanding this.